src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/AttributeExtensionPass.php

```diff
@@ -33,6 +36,14 @@ final class AttributeExtensionPass implements CompilerPassInterface
 
     public static function autoconfigureFromAttribute(ChildDefinition $definition, AsTwigFilter|AsTwigFunction|AsTwigTest $attribute, \ReflectionMethod $reflector): void
     {
+        $class = $reflector->getDeclaringClass();
+        if ($class->implementsInterface(ExtensionInterface::class)) {
+            if ($class->isSubclassOf(AbstractExtension::class)) {
+                throw new LogicException(sprintf('The class "%s" cannot extend "%s" and use the "#[%s]" attribute on method "%s()".', $class->name, AbstractExtension::class, $attribute::class, $reflector->name));
+            }
```

Comment:
Checking `AbstractExtension` makes the error message more specific to the code, as most extensions extends this class and doesn't implement the interface directly.